src/coreclr/jit/morph.cpp

```diff
@@ -787,17 +784,20 @@ void CallArgs::ArgsComplete(Compiler* comp, GenTreeCall* call)
 
         if (argx->gtFlags & GTF_ASG)
         {
-            // If this is not the only argument, or it's a copyblk, or it already evaluates the expression to
-            // a tmp, then we need a temp in the late arg list.
-            if ((argCount > 1) || argx->OperIsCopyBlkOp()
-#ifdef FEATURE_FIXED_OUT_ARGS
-                || arg.m_isTmp // Protect this by "FEATURE_FIXED_OUT_ARGS" to preserve the property
-                               // that we only have late non-register args when that feature is on.
-#endif
```

Comment:
`FEATURE_FIXED_OUT_ARGS` is defined even on x86, just with value 0, so this `#ifdef` was always true. The point of this check is to catch when `fgMakeOutgoingStructArgCopy` has left a non-value node, and it only does this for `FEATURE_FIXED_OUT_ARGS`, so I have fixed the logic.